File: zig/resistor-color/resistor_color.zig
const std = @import("std");

pub const ColorBand = enum {
    black,
    brown,
    red,
    orange,
    yellow,
    green,
    blue,
    violet,
    grey,
    white,
};

/// Returns the resistance value (ordinal from its enum) based on the given `color`.
pub fn colorCode(color: ColorBand) usize {
    return @intFromEnum(color);
}

/// Returns the set of all colors in `ColorBand`
pub fn colors() []const ColorBand {
    return std.enums.values(ColorBand);
}

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;

class ResistorColor {
    int colorCode(String color) {
        return Arrays
                .stream(Colors.values())
                .filter(c -> c.name().equalsIgnoreCase(color))
                .findAny()
                .get()
                .ordinal();
    }

    String[] colors() {
        return Arrays
                .stream(Colors.values())
                .map(Colors::name)
                .toArray(String[]::new);
    }

    enum Colors {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white,
    }
}
